### PR TITLE
feat(config): support EcoDim dimmers

### PR DESCRIPTION
**Pull request by pascalwinters**
Adds support for EcoDim 0.7 Basic variant and optimized EcoDim 0.7 configuration.

### CODE REVIEW

**Comment by blhoward2** [comment]
@pascalwinters From where did you get these parameters? The zwave alliance shows two devices, the DIM.10 and the DIM.7. You're calling this the DIM.07 but have the product ID for the DIM.10 (0x0002) instead of the DIM.07 (0x0001). You're also including the parameters for the DIM.10, which include many more options than the DIM.07. From where are you getting your ProductID and that it is the DIM.07? What is your firmware version?

What you're calling the DIM.07 Basic is actually the DIM.10 in the database. This may be a DIM.07 with new firmware that now supports more parameters. They may sell the DIM.07 (some variant) and the DIM.10 that report the same info, but if so the device file should reflect that. And the original one (0x0001) should be called just the DIM.07.

Edit: I suspect from googling that this is a firmware split or multiple devices using the same Product ID. The DIM.07 has the shorter parameter list. The Dim.07 Pro does show the parameters from the DIM.10. But unless you can verify it I'm not sure this is the correct Product ID. If it is, the shorter one should be the DIM.07 and this one should be the DIM.07 Pro / Dim.10. If it isn't, it may be that the two DIM.07s have the same product ID but at a specific firmware version they pickup the extra options. Make sense? In any event this is wrong as what we’re calling the DIM.07 Pro isn’t that device and that device had many more parameters available. 

DIM.07: https://products.z-wavealliance.org/products/3796

DIM.10: https://products.z-wavealliance.org/products/3932
https://www.ecodim.nl/eco-dim10-zwave.html

@AlCalzone Hold off on merging this one as I'm not sure that it is correct.

**Comment by pascalwinters** [comment]
@blhoward2 I have all the info from the ZWaveJS2MQTT dashboard:
image
Here you can see the ProductId 0x002, and I have an ECO-DIM.07 Basic variant.
I had contact with the manufacturer yesterday, the basic variant isn't yet online on zwavealliance, it's a matter of time.

I copied the parameters from the 07 Pro version, because it's a copy of that device. But maybe it's better to contact the supplier and ask for all information and process it in de config files. Maybe I can then also add the 10 variant, and I can also the mystery x0202 productid.

I will contact the supplier to clear up everything.

**Comment by AlCalzone** [comment]
@pascalwinters some manufacturers reuse the same device IDs over multiple firmwares. It might be that we have to distinguish the devices by firmware version. Any chance you see your version anywhere? If not, you could send me your cache files and I'll check.

**Comment by pascalwinters** [comment]
@AlCalzone I have firmware version 1.12.
Is there an central place to see all zwave firmware versions?

**Comment by AlCalzone** [comment]
You mean in zwavejs2mqtt? I don't think so. @robertsLando ?

**Comment by robertsLando** [comment]
> Is there an central place to see all zwave firmware versions?

Could be good to add the fw version in the table? @pascalwinters

**Comment by pascalwinters** [comment]
@AlCalzone No not especially in zwavejs2mqtt, but how and where can you see if there is an new firmware available?

**Comment by pascalwinters** [comment]
> > Is there an central place to see all zwave firmware versions?
> 
> Could be good to add the fw version in the table? @pascalwinters

Which table do you mean? @robertsLando 
It's already available in the zwavejstmqtt:
image

**Comment by robertsLando** [comment]
@pascalwinters I mean the main table, where you see all your nodes

**Comment by pascalwinters** [comment]
> @pascalwinters I mean the main table, where you see all your nodes

@robertsLando Maybe it can, but it's not part of this PR

**Comment by AlCalzone** [comment]
> where can you see if there is an new firmware available?

That's not possible

**Comment by pascalwinters** [comment]
I have requested the info by the manufacturer. I will edit this PR when I received the info.

**Comment by pascalwinters** [comment]
The manufacturer indicates that the chips used in the 3 dimmers are all the same and can therefore all have the same product ID. 
@AlCalzone How do we handle this?

I have replied with the question if we can detect it with the firmware versions.

**Comment by AlCalzone** [comment]
That is a weird explanation. Let's wait for info on the firmware.

**Comment by pascalwinters** [comment]
Ok, we’ll wait. Otherwise I will call them tomorrow.

File: packages/config/config/devices/0x0431/ecodim.json [hunk 1]
```diff
@@ -101,29 +110,29 @@
 			"description": "Enable or Disable external switch to pair network. Default value: 0.",
 			"valueSize": 1,
 			"minValue": 0,
-			"maxValue": 255,
-			"defaultValue": 0,
+			"maxValue": 1,
+			"defaultValue": 1,
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0431/ecodim.json]: Description says "default 0", line 114 says "default 1".

File: packages/config/config/devices/0x0431/ecodim.json [hunk 2]
```diff
@@ -101,29 +110,29 @@
 			"description": "Enable or Disable external switch to pair network. Default value: 0.",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0431/ecodim.json]:
```suggestion
			"description": "Enable or Disable external switch to pair network.",
```
I prefer not having default values or value ranges in the description.

File: packages/config/config/devices/0x0431/ecodim.json [hunk 3]
```diff
@@ -78,13 +83,17 @@
...
+				{
+					"label": "Send Switch Multilevel Report only when Load status is not changed by Z-WAVE Command",
+					"value": 2
 				}
 			]
 		},
 		"3": {
-			"label": "Dimming speed",
+			"label": "Default dimming speed",
 			"description": "Default dimming speed，unit is second. Valid values: 0-100. Default value: 5 0 - Instanly. 1 - Fast. ... 100 - Slow.",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0431/ecodim.json]:
```suggestion
			"description": "Default dimming speed",
```
but that is rather pointless

**Review by AlCalzone — CHANGES_REQUESTED** [review]

File: packages/config/config/devices/0x0431/ecodim.json (hunk 3 above)
Comment (pascalwinters) [reply packages/config/config/devices/0x0431/ecodim.json]: Yes agree, but it's from the manual

File: packages/config/config/devices/0x0431/ecodim.json (hunk 1 above)
Comment (pascalwinters) [reply packages/config/config/devices/0x0431/ecodim.json]: Changed it

**Review by AlCalzone — APPROVED** [review]

**Pull request merged**